### PR TITLE
Add tasks for German Embedding Evaluation

### PR DESCRIPTION
**Pull request by guenthermi**
We added a few tasks to evaluate German embedding models:

**PawsX** (https://huggingface.co/datasets/paws-x): This dataset contains pairs of an original text and a paraphrased version of it. The labels determine whether it is a real paraphrase or the content has a different semantic. We add this as a pair classification task to MTEB

**MIRACL** (https://huggingface.co/datasets/miracl/miracl): This is originally retrieval dataset. It contains a corpus of passages from Wikipedia articles and human-annotated relevance judgements for some of them.  We only took passages where relevance judgements are available since those are suitable for an MTEB re-ranking task.

**GerDaLIR** (https://github.com/lavis-nlp/GerDaLIR): GerDaLIR is a legal information retrieval dataset created from the Open Legal Data platform with the same name.

**GermanDPR** (https://www.deepset.ai/germanquad): GermanDPR is a passage retrieval tasks for question answering, i.e. it contains questions associated to passages which contain the relevant answer.

**XMarket** (https://xmrec.github.io/): This is originally a ecommerce product dataset which contains titles and descriptions of products which are associated to very fine-granular categories. We created a category to product description retrieval task out of it.

**GermanSTSBenchmarkSTS** (https://github.com/t-systems-on-site-services-gmbh/german-STSbenchmark): This dataset is an STS dataset created by T-Systems. It contains translations of English STS datasets.

---

I also updated the script in `scripts/run_mteb_german.py` to include those additional tasks, as well as the exiting German tasks for clustering and STS

### CODE REVIEW

**Comment by guenthermi** [comment]
Seems also be related to #183

**Comment by yvesloy** [comment]
Do you already have some results on what embedding model works best for german atm?

**Comment by guenthermi** [comment]
> Do you already have some results on what embedding model works best for german atm?

I have shared some results here:
https://twitter.com/michael_g_u/status/1747293709849227618

**Comment by guenthermi** [comment]
@Muennighoff is it possible to assign someone to review this PR?

**Comment by Muennighoff** [comment]
> [guenthermi](/guenthermi)

Taking a look! Also invited you to the org so you can request reviews in the future 👍
Amazing work on this!

File: mteb/tasks/STS/__init__.py [hunk 1]
```diff
@@ -10,3 +10,4 @@
 from .STSBenchmarkSTS import *
 from .CMTEBSTS import *
 from .PolishSTS import *
+from .GermanSTSBenchmarkSTS import *
```
Comment (Muennighoff) [inline mteb/tasks/STS/__init__.py:13]:
```suggestion
from .CMTEBSTS import *
from .GermanSTSBenchmarkSTS import *
from .PolishSTS import *
from .STSBenchmarkSTS import *
```

File: mteb/tasks/Reranking/MIRACLReranking.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import datasets
+from mteb.abstasks.AbsTaskReranking import AbsTaskReranking
```
Comment (Muennighoff) [inline mteb/tasks/Reranking/MIRACLReranking.py:2]:
```suggestion
from ...abstasks.AbsTaskReranking import AbsTaskReranking

import datasets
```

**Review by Muennighoff — APPROVED** [review]
LGTM amazing work!

**Comment by Muennighoff** [comment]
Let's merge?

**Comment by guenthermi** [comment]
> Let's merge?

It did't work out with just adding more languages for PawsX so I had to adjust the code in AbsTaskPairClassification to support multilingual datasets. If it looks ok to you, we can merge it

I evaluated some multilingual e5 models on the PawsX task:

|                       |       de |       en |       es |       fr |       ja |       ko |       zh |
|:----------------------|---------:|---------:|---------:|---------:|---------:|---------:|---------:|
| e5-multilingual-small | 0.538849 | 0.548835 | 0.537453 | 0.556763 | 0.493585 | 0.512998 | 0.54783  |
| e5-multilingual-base  | 0.544398 | 0.540388 | 0.550975 | 0.569298 | 0.497352 | 0.511751 | 0.556259 |
| e5-multilingual-large | 0.575009 | 0.595169 | 0.563678 | 0.585035 | 0.49574  | 0.516579 | 0.57002  |

**Pull request merged**